### DISCUSSION TITLE
Comparing SEBE algorithm with real electrical production

### DISCUSSION THREAD

**Discussion by Andrei99537** [UMEP-dev/UMEP#627] [Q&A]
Hello,
I have a question regarding a solar analysis I conducted to compare the accuracy of the SEBE algorithm with ArcGIS's Area Solar Radiation and actual PV electrical production. To obtain the results, I used a DSM, meteorological file, and shapefiles representing only the rooftop areas where solar PVs are installed.

Using ArcGIS's solar radiation algorithm, I obtained data that I converted to MWh/year of electrical power production. However, when I used the SEBE algorithm, I got the final results in kWh/year (I believe) during the last step, "Zonal statistics", but the difference between the two results and the actual electrical production is significant. Do you have any ideas on what conversions I should perform to obtain similar values using the SEBE algorithm? *If i try to convert from Kwh to Mwh it doesn't help*
 
As example, this is the actual value of total electricity produced for one of the building where PV panels are planted:
      
![image](https://github.com/UMEP-dev/UMEP/assets/63879280/6afb4aa1-41ef-4cef-96c6-64079b19af1e)

 This is the estimation of electrical production power using the ArcGis's algorithm (MWh/year):
  
![image](https://github.com/UMEP-dev/UMEP/assets/63879280/b883127f-9202-4a2d-92d9-5d8e6e32713d)

And this is the result that i obtain by using the SEBE algorithm (Kwh):

![image](https://github.com/UMEP-dev/UMEP/assets/63879280/6a52b46c-44d7-42b0-9626-f073d3bee206)

Also, the point value of the Energy_mp_slope65_roof raster seem to be a little higher

![image](https://github.com/UMEP-dev/UMEP/assets/63879280/65770b2f-eacd-44cd-860f-d9b4831daebe)

Thank you so much!

**Comment by biglimp** [comment]
Interesting comparison. SEBE output is kWh/year if a metfile that is one year long is used. If you have multiple years in your metfile, the model calculates an average still giving you kWh/year. If you are using a shorter time length in your metfile, you get values for that period only. The file is still called Energyyearroof.tif (un-pedagogic from our side). 

How do you convert irradiance to energy production?

Maybe you can attach your final result and the metfile for us to have a look at.

**Reply by Andrei99537** [reply]
I am using only DSM, 1 year meteo file (1985 unfortunately) and PV's shapefile and i still get a sum of 1309313.43420 kwh and an mean of 338.1451 Kwh/year. It seems like any conversion i do in order to obtain electrical production does not work, obtaining only 25.83 Kwh/year or other values (as shown belor), and compared to the actual value it won't do the trick.
![image](https://github.com/UMEP-dev/UMEP/assets/63879280/633d55ec-0bbc-4715-89cc-1ad1d0b01533)

Do you know any parameters or conversion methods that i could apply in order to obtain accurate values from Energy_mp_slope65 raster or any other rasters that have been generated along the process by the SEBE algorithm?

**Reply by biglimp** [reply]
Attach your input and output. It will be easier for me to interpret.

**Reply by Andrei99537** [reply]
[LSB_data.zip](https://github.com/user-attachments/files/15792530/LSB_data.zip)

**Reply by biglimp** [reply]
Hm, there is something wrong here. I suspect that your CRS is causing issues. Can you provide me with your raw output (Energyroofyear.tif) and the textfile RunInfoSEBE.txt which should be olcated in your output folder as well.

**Reply by Andrei99537** [reply]
This should be it

[Energyyearroof_runinfo.zip](https://github.com/user-attachments/files/15804329/Energyyearroof_runinfo.zip)

**Reply by biglimp** [reply]
Something must have gone wrong when you ran the model. I just tried to run with your data for the DSM that you send. Here is what it should look like. Now I get around 1950 kWh on the roof that you are investigating Try to run the model again. Maybe you switsched aspect and height raster?

![image](https://github.com/UMEP-dev/UMEP/assets/11908356/6a5d78f9-66ed-45b0-a5d5-66df8a4d6cfc)

**Reply by Andrei99537** [reply]
I ran the model again and indeed, now i get a mean of 2299.565 Kwh/year. i tried to convert it in energy, by multiplying the area * mean / 1000 (in order to obtain Mwh/year) to extract usable area, then i used the follow statement as electricity production:

"The United States Environmental Protection Agency (EPA) [provides a conservative best estimate](https://www.epa.gov/green-power-markets/green-power-equivalency-calculator-calculations-and-references) of 16 percent efficiency and 86 percent performance ratio. These values mean that the solar panels are capable of converting 16 percent of incoming solar energy into electricity, and then 86 percent of that electricity is preserved as it goes through the installation. (ESRI)", which means "Usable_irad_area * 0.16 * 0.86

My final value of electricity production is 196.020 Mwh/year for building on top of the map, which is a value that is more than double of the true quantity. 

However, the fact is that the total amount of electricity produced by all the PV panels in my area of study matches the total amount of electricity that is produced for all of the buildings in my area for the whole year >>Refference: 519.44 Mwh/year; My calculation: 524.45 Mwh/year (even tho the quantities of electricity per building do not match at all) 

Any sugestions or opinions regarding what could i do different?

Thank you so much for your time!

**Reply by biglimp** [reply]
No real good ideas here. If the values from SEBE match other models (ArcGIS and GRASS), then it must be the area and electricity production conversion that is the problem. Sorry, I dont know how I can assist you further.
